File: 5-helo/helo.php
#!/usr/bin/env -S -- php
<?php
chdir(__DIR__);

$f = basename(__FILE__);
printf("HELO :: VIA -- {$f}\n");

$sh = join(
  " ",
  array_map("escapeshellarg", [
    "bat",
    "--color=always",
    "--decorations=always",
    "--",
    __FILE__,
  ])
);
$code = -1;
assert(passthru($sh, $code) == null);
assert($code == 0);


?>

File: 5-helo/helo.lua
#!/usr/bin/env -S -- lua -W --

local file = arg[0]
print("HELO :: VIA -- " .. file:match([[.*/(.+%.lua)]]))
local pf =
  assert(io.popen("bat --color=always --decorations=always -- " .. file, "r"))
local data = assert(pf:read("*a"))
pf:close()
print(data)
